### PR TITLE
OBPIH-5622 Fixed syntax error with SQL used to search products by pro…

### PR DESCRIPTION
**Pull request by jmiranda**
…duct type in inventory browser

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy [hunk 1]
```diff
@@ -761,12 +761,24 @@ class ProductAvailabilityService {
...
-                            add(Restrictions.conjunction().
-                                    add(Subqueries.lt(0, aggregatedQuantityQuery)).
-                                    add(Restrictions.in("productType", searchableProductTypes))
-                            ))
+
+                    // SUM(product availability.quantity_on_hand) > 0
+                    def quantityGreaterThanZero = Subqueries.lt(0, aggregatedQuantityQuery)
+                    // productType in (:searchableProductTypes)
+                    def inSearchableProductTypes = Restrictions.in("productType", searchableProductTypes)
+                    // productType in (:searchableProductTypes)
```
Comment (kchelstowski) [inline grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy:769]:
```suggestion
                    // productType in (:searchableNoStockProductTypes)
```

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy [hunk 2]
```diff
@@ -761,12 +761,24 @@ class ProductAvailabilityService {
...
+                    def inSearchableNoStockProductTypes = Restrictions.in("productType", searchableNoStockProductTypes)
+
+                    // Create a disjunction with none, one or both of the searchable product type restrictions
+                    def disjunction = Restrictions.disjunction()
+
+                    if (searchableNoStockProductTypes)
+                        disjunction.add(inSearchableNoStockProductTypes)
```
Comment (kchelstowski) [inline grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy:776]: I'm wondering how it works in Groovy if you do not add braces to `if` and you add some statements below? 
Does it only read up to one line below or?

**Review by kchelstowski — APPROVED** [review]
LGTM

File: grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy (hunk 2 above)
Comment (drodzewicz) [reply grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy:776]: I believe it works just like in `Java` and `C`. If the `if` statement doesn't have braces it only includes the first line in the under the `if(...)`.

For example if we have such a statement
```
if (condition == true)
    println "first print"
    println "second print"
```
Even though based on code formatting it looks like the second print belongs to the `if` statement in reality it is outside of the block.

**Pull request merged**